### PR TITLE
Make test should pass locally

### PR DESCRIPTION
**Pull request by ZackLK**
- Fix render_test when running locally - tests can't depend on dev's local timezone.
- Add test flags that are used to skip tests that have external dependencies. 
-- Plumb the RequiresX() functions into tests that depend on them.

Make test now runs failure-free locally.

### CODE REVIEW

File: testflags/testflags.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+func init() {
+	flag.BoolVar(&TestFlags.mysql, "mysql", false, "MySQL external dependency")
+	flag.BoolVar(&TestFlags.postgres, "postgres", false, "PostGreSQL external dependency")
+	flag.BoolVar(&TestFlags.mongodb, "mongodb", false, "MongoDB external dependency")
+	flag.BoolVar(&TestFlags.cassandra, "cassandra", false, "Cassandra external dependency")
+}
+
+// TODO: Better instructions:
+// 1) How to start external dependencies - maybe link to docs so we can change it easily?
```
Comment (Groxx) [inline testflags/testflags.go:45]: probably going to be something like "start cassandra (very likely `docker run cassandra:version`), and install the schema".
but that can come later.  and there's a million acceptable ways to do it, so deciding what's best may take a few rounds.

File: testflags/testflags.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package testflags
```
Comment (Groxx) [inline testflags/testflags.go:23]: I like it better than any other name I've thought of 👍  Very clear what it's looking for / related to.

File: host/testcluster.go [hunk 3]
```diff
@@ -200,11 +202,14 @@ func NewPersistenceTestCluster(clusterConfig *TestClusterConfig) testcluster.Per
 		ops := clusterConfig.Persistence
 		ops.DBPluginName = "cassandra"
 		testCluster = nosql.NewTestCluster(ops.DBPluginName, ops.DBName, ops.DBUsername, ops.DBPassword, ops.DBHost, ops.DBPort, ops.ProtoVersion, "")
+		testflags.RequireCassandra(t)
```
Comment (Groxx) [inline host/testcluster.go:205]:
```suggestion
		testflags.RequireCassandra(t)
		testCluster = nosql.NewTestCluster(ops.DBPluginName, ops.DBName, ops.DBUsername, ops.DBPassword, ops.DBHost, ops.DBPort, ops.ProtoVersion, "")
```

File: tools/cli/render_test.go [hunk 4]
```diff
@@ -182,7 +182,7 @@ var testTable = []testRow{
 		StringField: "very long/ string this is",
 		IntField:    456,
 		BoolField:   false,
-		TimeField:   time.Date(2000, 11, 12, 13, 14, 15, 16, time.Local),
+		TimeField:   time.Date(2000, 11, 12, 13, 14, 15, 16, time.UTC),
```
Comment (Groxx) [inline tools/cli/render_test.go:185]: heh.  yeah, or some other fixed time zone if we want to assert that it localizes.
UTC is fine / consistent tho.

**Review by Groxx — COMMENTED** [review]
A bit of a shame that some are pushed into a base-test-helper-like-thing and some are not... but I don't think it's necessary to be consistent there.  some packages are more clearly "depends on external stuff" than others.

**Comment by ZackLK** [comment]
I made the Makefile / buildkite changes and have ensured no tests are being skipped in buildkite, by downloading before and after test logs and comparing them. No tests are skipped in buildkite, and the exact same set of tests were run before and after my changes.

File: Makefile [hunk 5]
```diff
@@ -472,7 +472,7 @@ INTEG_TEST_XDC_ROOT=./host/xdc
 INTEG_TEST_XDC_DIR=hostxdc
 INTEG_TEST_NDC_ROOT=./host/ndc
 INTEG_TEST_NDC_DIR=hostndc
-OPT_OUT_TEST=./bench/% ./canary/%
+OPT_OUT_TEST=
```
Comment (Groxx) [inline Makefile:475]: bleh.  I didn't notice this one before D:

**Review by Groxx — APPROVED** [review]

**Pull request merged**